shared/pingpong/accounts.go

```diff
@@ -186,133 +194,156 @@ func prepareAssets(accounts map[string]uint64, client libgoal.Client, cfg PpConf
 			totalSent++
 			throttleTransactionRate(startTime, cfg, totalSent)
 		}
-		account, accountErr := client.AccountInformation(addr)
-		if accountErr != nil {
-			fmt.Printf("Cannot lookup source account %v\n", addr)
-			err = accountErr
-			return
-		}
+	}
 
+	// wait until all the assets created
+	allAssets := make(map[uint64]string, cfg.NumAsset)
+	for addr := range accounts {
+		var account v1.Account
+		start := time.Now()
+		for {
+			account, err = client.AccountInformation(addr)
+			if err != nil {
+				fmt.Printf("Warning, cannot lookup source account")
+				return
+			}
+			if len(account.AssetParams) >= numCreatedAssetsByAddr[addr] {
+				break
+			}
+			if time.Now().After(start.Add(3 * time.Minute)) {
+				err = fmt.Errorf("asset creation took too long")
+				fmt.Printf("Error: %s\n", err.Error())
+				return
+			}
+			time.Sleep(time.Second)
```

Comment:
Sleeping here is not what you really want to do. instead, you want to track the current round number and call `WaitForRound`. Before the round would change, there is no reason to keep polling the asset information.

Reply:
done